packages/core/src/runtime/Browser.ts

```
@@ -600,4 +601,143 @@ export class Browser<T> implements BrowserInterface {
 	public async close(): Promise<void> {
 		await this.client.browser.close()
 	}
+
+	// Check the target has type Locator or not
+	private isLocator(target: Locator | ElementHandle | Point | ScrollDirection): target is Locator {
+		return !Array.isArray(target) && (target as Locator).find !== undefined
+	}
+
+	// Check the target has type ElementHandle or not
+	private isElementHandle(
+		target: Locator | ElementHandle | Point | ScrollDirection,
+	): target is ElementHandle {
+		return (target as ElementHandle).element !== undefined
+	}
+
+	private isPoint(target: Locator | ElementHandle | Point | ScrollDirection): target is Point {
+		return (
+			Array.isArray(target) &&
+			target.length === 2 &&
+			typeof target[0] === 'number' &&
+			typeof target[1] === 'number'
```

is it enough?
```suggestion
			Array.isArray(target) &&
			typeof target[0] === 'number' &&
			typeof target[1] === 'number'
```

---

To prevent the end-user using `[1, 2, 'hello world']` or something like this.

---

in this case, the target is Point, so it should be `[number, number]`, the compile will fail if the input is `[1, 2, 'hello world']` 🤔

---

I want to check the length because I want to throw a friendly error message to the end-user if they input the incorrect type of param. The compile will fail but maybe the end-user will not get a friendly error message.

---

okay

---

Resolved.